arch/arm/src/armv7-a/arm_cache.c

```diff
@@ -189,7 +162,14 @@ void up_disable_icache(void)
 
 size_t up_get_dcache_linesize(void)
 {
-  return up_get_cache_linesize();
+  static uint32_t clsize;
+
+  if (clsize == 0)
+    {
+      clsize = MAX(cp15_dcache_linesize(), l2cc_linesize());
```

Comment:
Should we include some DEBUG_ASSERTION in fact the clsize is invalid (i.e. 0 or some value too big that doesn't make sense?)

Reply:
LGTM! I'm going to do a check on the results after  get the clsize.